src/display.jl

```diff
@@ -204,6 +204,40 @@ function basic_representation(a::Interval, format=nothing)
     output
 end
 
+function basic_representation(a::Interval{Float32}, format=nothing)
```

Comment:
You are duplicating a lot of code. You should be able to do this by multiple dispatch instead for just the bits that change depending on whether you have Float64 or Float32 or BigFloat.